Including env vars in octostache substitution (#2)
im-open/octostache-action: README.md

# authenticate-with-gh-package-registries

> This action will scan the file(s) provided in the `file-with-substitutions` argument for Octopus variable substitution syntax `#{VariableName}`.  If it finds a matching variable in the `variables-file` or in the environment variables, it will replace the template with the actual value.

Reviewer: I think I wrote this summary initially and now I think it needs some clarification...
```suggestion
This action will scan the file(s) provided in the `files-with-substitutions` argument for Octopus variable substitution syntax `#{VariableName}`.  If the files contain any `#{Variables}` that match an item in the `variables-file` or environment variables, it will replace the template with the actual value.
```

This is a container action so it will not work on Windows runners.

## Inputs

| Parameter                  | Is Required | Description                                                                                                                                                                                                                                              |
| -------------------------- | ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `variables-file`           | false       | An optional yaml file containing variables to use in the substitution.                                                                                                                                                                                   |

> | `files-with-substitutions` | true        | A comma separated list of files that contains the octo tokens needing to be substituted.                                                                                                                                                                 |

Reviewer: I don't know if you are tied to this wording but something like this might be a little smoother:
```suggestion
| `files-with-substitutions` | true        | A comma separated list of files with `#{variables}` that need substitution.                                                                                                                                                                 |
```

> | `output-files`             | false       | An optional comma separated list of files to output.<br/>If defined, the program assumes the index of the output file is the same as the index of the template file in the template-files list. They therefore need to have the same number of elements. |

Reviewer: I don't think I'm following this explanation, it's also referring to them as template files which the program uses but we're input it as files-with-substitutions.

Author: I had a hard time thinking of how to describe this behavior. Basically I want to say that if `output-files` is defined, it is assumed that each index corresponds to the matching index of the `files-with-substitutions` list.

For example if we got the following inputs
`files-with-substitutions`: ./file1.js, ./src/anotherfile.js, ./src/appsettings.config
`output-files`: ./file1.js, ./file2.js, ./appsettings.config

and were processing ./src/anotherfile.js, the outputted file would be ./file2.js. If you can think of a better way to word that, which I'm sure there is, please let me know!

### Variables File

```yaml
Environment: Dev
Version: 1.3.62

> LaunchDarklyKey: abc

Reviewer: Do you want to keep LaunchDarklyKey here since you're providing it as an environment variable in the example?

Author: Yeah, I wanted to show that the environment variable would override the file. I don't know if I actually mentioned that behavior in the readme though...I'll have to add it if it isn't there already.

>       - uses: im-open/octostache-action@v1.0.0

Reviewer: ```suggestion
      - uses: im-open/octostache-action@v2.0.0
```

        with:
          variables-file: ./substitution-variables.json
          files-with-substitutions: ./src/DemoApp19/DemoApp19.csproj,./src/DemoApp19/Bff/FrontEnd/scripts/build-variables.js

>           output-files: ./src/DemoApp19/DemoApp19.csproj,./src/DemoApp19/Bff/FrontEnd/scripts/build-variables.js

Reviewer: I'm curious about this and it's something I wondered about with the original action.  Is there a compelling reason to allow it to output to a different file?  Would it simplify things if we just said the action reads/writes to the same file?

@izep

Author: Having the separate output files makes testing easier for sure, I'll say that much. Other than that 🤷

        env:

>           LaunchDarklyKey: ${{ secrets.LAUNCH_DARKLY_API_KEY }}

Reviewer: This might be a scenario you want to cover in the readme, what takes precedence if you have it in the file and in an environment variable.
